src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/SdkClient/ResourceManagerSdkClient.cs

```diff
@@ -312,23 +312,33 @@ public static List<string> ParseDetailErrorMessage(string statusMessage)
             Action<string, string, Deployment> job,
             params ProvisioningState[] status)
         {
-            DeploymentExtended deployment;
+            DeploymentExtended deployment = null;
+            bool deploymentExists = false;
             int counter = 5000;
 
-            do
+            while (deployment == null ||
```

Comment:
Please ignore this change for now. I did some more investigation on this and I'm refactoring this change. Will update soon.